packages/terra-doc-template/src/DocTemplate.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import IndexExampleTemplate from './IndexPageTemplate';
+
+const propTypes = {
+  /**
+   * The given component's package name
+   */
+  packageName: PropTypes.string,
+  /**
+   * The given component's readme file imported to a string
+   */
+  readme: PropTypes.string,
+  /**
+   * All of the example(s) that will be displayed. An empty array is supported.
+   * ```
+   * title: The title of the example
+   * description: A description of the example to be displayed below the title
+   * example: The example to be displayed
+   * source: The source code of the example
+   * ```
+   */
+  examples: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        title: PropTypes.string,
+        description: PropTypes.string,
+        example: PropTypes.element,
+        source: PropTypes.string,
+      },
+    ),
+  ),
+  /**
+   * All of the props table(s) that will be displayed. An empty array is supported.
+   * ```
+   * componentSrc: The source code of the component
+   * componentName: The name of the component
+   * ```
+   */
+  propsTables: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        componentSrc: PropTypes.string,
+        componentName: PropTypes.string,
+      },
+    ),
+  ),
+};
+
+const defaultProps = {
+  packageName: '',
+  readme: '',
+  examples: [],
+  propsTables: [],
+};
```

Comment:
Any props that are expected to be supplied within the DocTemplate could be updated to be marked as required. And any required props can be removed from the default props. 

```
packageName: PropTypes.string.isRequired
```

Reply:
The only prop I think we might want to make required would be the readme. But if someone just wants to display some examples, they wouldn't want to have to add a readme. I'm open to hearing what others think.

Reply:
If we made any prop required, I feel the package name makes the most sense. However I don't think it is bad to leave the template open/flexible